mlrun/package/packagers/pandas_packagers.py

```diff
@@ -94,27 +193,452 @@ class _CSVFormatter(_Formatter):
     """
 
     @classmethod
-    def to(cls, obj: pd.DataFrame, file_path: str, **to_kwargs: dict):
+    def to(
+        cls, obj: pd.DataFrame, file_path: str, flatten: bool = True, **to_kwargs
+    ) -> dict:
         """
-        Save the given dataframe / series to the file path given.
+        Save the given dataframe to the csv file path given.
 
-        :param obj:       The dataframe / series to save.
+        :param obj:       The dataframe to save.
         :param file_path: The file to save to.
-        :param to_kwargs: Additional keyword arguments to pass to the relevant `to_csv` function.
+        :param flatten:   Whether to flatten the dataframe before saving. For some formats it is mandatory to enable
+                          flattening, otherwise saving and loading the dataframe will cause unexpected behavior
+                          especially in case it is multi-level or multi-index. Default to True.
+        :param to_kwargs: Additional keyword arguments to pass to the `to_csv` function.
+
+        :return A dictionary of keyword arguments for reading the dataframe from file.
         """
+        # Flatten the dataframe (this format have problems saving multi-level dataframes):
```

Comment:
do you have examples/refrences of when it's problematic to not flatten? consider logging a warning in case of not flattening.

Reply:
Formats like `feather` and `orc` can't save multi-level columns and indexes. And, text formats like `json`, `html`, `csv`, `xml` and `xlsx` don't handle it properly, leaving a messy file with weird naming for columns and indexes. When I flatten the dataframe I make sure to give the user back the dataframe as it was. In my eyes a person who even once saved a regular pandas as csv and got the default index column as `"Unnamed: 0"` will know what I'm talking about.